cmds/exp/madeye/madeye.go

```diff
@@ -0,0 +1,144 @@
+// Copyright 2013-2020 the u-root Authors. All rights reserved
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+// madeye merges u-root initramfs to form a single universal initramfs.
+// Synopsis:
+//     madeye initramfs [initramfs...]
+//
+// Description:
+package main
+
+import (
+	"bytes"
+	"flag"
+	"fmt"
+	"io"
+	"io/ioutil"
+	"log"
+	"os"
+	"path/filepath"
+
+	"github.com/u-root/u-root/pkg/cpio"
+	"github.com/u-root/u-root/pkg/uio"
+	"golang.org/x/sys/unix"
+)
+
+var (
+	debug = func(string, ...interface{}) {}
+	d     = flag.Bool("v", false, "Debug prints")
+	arch  = map[string]string{
+		"initramfs.linux_amd64.cpio": "amd64",
+		"initramfs.linux_arm.cpio":   "arm",
+	}
+	out = map[string]*cpio.Record{}
+)
+
+func usage() {
+	log.Fatalf("Usage: madeye")
+}
+
+func file(archiver cpio.RecordFormat, n string, f io.ReaderAt) ([]cpio.Record, error) {
+	var r []cpio.Record
+	rr := archiver.Reader(f)
+	a, ok := arch[filepath.Base(n)]
+	if !ok {
+		return r, fmt.Errorf("%s: don't know about this", n)
```

Comment:
does this mean you can only process arch-specific initrds?  (versus a generic one in addition to an arch one)

Reply:
for now. I think at some point we may want to merge other stuff too.